### PR TITLE
166840926 article comments

### PR DESCRIPTION
**Pull request by olorunwalawrence**
#### What does this PR do?
create article comment
#### Description of the task to be completed
- create article comment functionality
- create the endpoint for the article
#### How to manually test this PR
- clone the [repo ](https://github.com/andela/ah-kifaru-backend)
- on your terminal, run ` 166840926-article-comments`
- create a user model and migration
- on your terminal, run npm install to install dependencies
- on your terminal, run npm run start:dev to start the application.

#### What is the story identity?

166840926

### CODE REVIEW

**Review by tohbay — APPROVED** [review]
Good job @olorunwalawrence 👍

File: src/test/article.spec.js [hunk 1]
```diff
@@ -1197,5 +1199,255 @@ describe('GET /api/v1/articles/popular', () => {
...
+    const fakeToken = '12345678iokjhgfdsdfghjk';
+    const res = await server()
+      .get(`/api/v1/comments/${createdArticle.id}`)
+      .set('token', fakeToken);
+    expect(res.status).to.equal(401);
+    expect(res.body.message).to.be.equal('Token is not valid');
+  });
+});
+
+describe('PUT api/v1/comments/:commentId', async () => {
```
Comment (timi-codes) [inline src/test/article.spec.js:1389]: Hello Good @olorunwalawrence. However, None of your tests here catered for a successful comment update. Kindly revisit this.

**Review by timi-codes — CHANGES_REQUESTED** [review]
Good Job! @lawarence, I left you a comment. Kindly review and let's get the merged as soon as possible

File: src/test/article.spec.js [hunk 2]
```diff
@@ -1197,5 +1199,255 @@ describe('GET /api/v1/articles/popular', () => {
...
+    const token = helper.jwtSigner({ id: comenter.id });
+    const res = await server()
+      .get(`/api/v1/comments/${createdArticle.id}`)
+      .set('token', token);
+    expect(res.status).to.equal(200);
+    expect(res.body.message).to.be.equal('Operation Successful');
+    expect((res.body.data.rows[0] = 1));
+  });
+
+  it('should not list  when comment token is invalid', async () => {
```
Comment (segunolalive) [inline src/test/article.spec.js:1375]: This behaviour is not right. Users should be able to see comments whether or not they're logged in.

File: src/test/article.spec.js [hunk 3]
```diff
@@ -1197,5 +1199,255 @@ describe('GET /api/v1/articles/popular', () => {
...
+    const { id: commentId, content } = createdComment[0].dataValues;
+    const fakeToken = 'wertyuiow3456789';
+    const res = await server()
+      .put(`/api/v1/comments/${commentId}`)
+      .set('token', fakeToken)
+      .send({ content });
+    expect(res.status).to.equal(401);
+    expect(res.body.message).to.be.equal('Token is not valid');
+  });
+  it('User should not be able to comment if not authenticated', async () => {
```
Comment (segunolalive) [inline src/test/article.spec.js]: should not be able to update a comment if not authenticated.

File: src/test/article.spec.js (hunk 1 above)
Comment (segunolalive) [reply src/test/article.spec.js:1389]: No success case in the tests. All the tests account for only for failure cases

**Review by segunolalive — COMMENTED** [review]
make the suggested changes including the grammatical errors in test descriptions.

This is good to go afterward.

**Pull request merged**